File: 프로그래머스/lv2/43165. 타겟 넘버/타겟 넘버.cpp
#include<bits/stdc++.h>
using namespace std;

int answer = 0;

void asd(vector<int> numbers, int target, int sum, int index)
{
    if (index == numbers.size())
    {
        if (sum == target)
        {
            answer++;
        }
        return;
    }

    asd(numbers, target, sum + numbers[index], index + 1);
    asd(numbers, target, sum - numbers[index], index + 1);

}


int solution(vector<int> numbers, int target) {

    asd(numbers, target, 0, 0);

    return answer;
}

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.java
class Solution {
    int answer = 0;

    public void cal(int[] numbers, int target, int sum, int index) {
        if (index == numbers.length) {
            if (sum == target) {
                answer++;
            }
            return;
        }
        cal(numbers, target, sum + numbers[index], index + 1);
        cal(numbers, target, sum - numbers[index], index + 1);
    }

    public int solution(int[] numbers, int target) {
        cal(numbers, target, 0, 0);
        return answer;
    }
}